tests/index_lifecycle.yaml

```diff
@@ -0,0 +1,85 @@
+$schema: ../../json_schemas/story.schema.yaml
+
+description: This story tests all endpoints relevant the lifecycle of an index, from creation to deletion.
+prologues: []
+epilogues:
```

Comment:
This story's epilogue will fail since movies doesn't exist and the games and books indices are being deleted as part of the chapters. Generally I think cleaning up after a test is the right thing because it avoids unnecessarily polluting the next test, and I suppose we can ignore the errors, but maybe there's a generic cleanup we can do between stories like delete all indices, snapshots, settings, etc.... just thinking out loud.

Reply:
Having stored procedures (borrowing a word from SQL land) that delete all indices and such is a brilliant idea. We should save it for a follow up PR though.

I think `ignore_errors` option is still necessaries for the cleanup to account for when not all chapters are run and we don't have a stored procedure that can clean up after said test.

Reply:
It's pretty unusual for a test framework. Generally you don't expect setup/teardown to fail with an unexpected error. You'll want to find a way where deleting an index succeeds on a 404 (index doesn't exist).

Reply:
I like this but don't like the name. I think rather than ignore, 404 means success in this case. So this is really something like 

```yaml
response:
  status:
    - 200
    - 404
```

Reply:
Just implemented `additional_success_statuses`
```
epilogues:
  - path: /books
    method: DELETE
    additional_success_statuses: [404]
```

Reply:
This hides that 200 status and doesn't allow you to remove 200. This would make it clearer:

```
epilogues:
  - path: /books
    method: DELETE
    status:
      - 200
      - 404
```